diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProvider.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProvider.java
index cb6a1fb31e6..e1136c3c0b6 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProvider.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProvider.java
@@ -125,13 +125,12 @@ public void testMetadata() throws Exception {
   @Test
   public void testOptions() throws Exception {
     Configuration conf = new Configuration();
-    conf.set(KeyProvider.DEFAULT_CIPHER_NAME, "myCipher");
-    conf.setInt(KeyProvider.DEFAULT_BITLENGTH_NAME, 512);
     Map<String, String> attributes = new HashMap<String, String>();
     attributes.put("a", "A");
     KeyProvider.Options options = KeyProvider.options(conf);
-    assertEquals("myCipher", options.getCipher());
-    assertEquals(512, options.getBitLength());
+    assertEquals(conf.get(KeyProvider.DEFAULT_CIPHER_NAME), options.getCipher());
+    assertEquals(conf.getInt(KeyProvider.DEFAULT_BITLENGTH_NAME, 0),
+        options.getBitLength());
     options.setCipher("yourCipher");
     options.setDescription("description");
     options.setAttributes(attributes);
