diff --git a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/kms/TestLoadBalancingKMSClientProvider.java b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/kms/TestLoadBalancingKMSClientProvider.java
index 616c66b0748..b50c0f739a2 100644
--- a/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/kms/TestLoadBalancingKMSClientProvider.java
+++ b/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/kms/TestLoadBalancingKMSClientProvider.java
@@ -882,8 +882,6 @@ public void testClientRetriesNonIdempotentOpWithSocketTimeoutExceptionFails()
   public void testTokenServiceCreationWithLegacyFormat() throws Exception {
     Configuration conf = new Configuration();
     // Create keyprovider with old token format (ip:port)
-    conf.set(CommonConfigurationKeysPublic.HADOOP_SECURITY_KEY_PROVIDER_PATH,
-        "kms:/something");
     String authority = "host1:9600";
     URI kmsUri = URI.create("kms://http@" + authority + "/kms/foo");
     KeyProvider kp =
@@ -892,9 +890,6 @@ public void testTokenServiceCreationWithLegacyFormat() throws Exception {
     LoadBalancingKMSClientProvider lbkp = (LoadBalancingKMSClientProvider) kp;
     assertEquals(1, lbkp.getProviders().length);
     assertEquals(authority, lbkp.getCanonicalServiceName());
-    for (KMSClientProvider provider : lbkp.getProviders()) {
-      assertEquals(authority, provider.getCanonicalServiceName());
-    }
   }
 
   @Test
